x-pack/plugins/security_solution/public/timelines/components/timeline/header/save_timeline.tsx

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiButton,
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFormRow,
+  EuiFlexItem,
+  EuiModal,
+  EuiModalBody,
+  EuiModalHeader,
+  EuiOverlayMask,
+  EuiSpacer,
+  EuiToolTip,
+  EuiProgress,
+} from '@elastic/eui';
+import React, { useCallback, useMemo } from 'react';
+import styled from 'styled-components';
+import { TimelineType } from '../../../../../common/types/timeline';
+import { useShallowEqualSelector } from '../../../../common/hooks/use_selector';
+import { timelineSelectors } from '../../../../timelines/store/timeline';
+import {
+  Description,
+  Name,
+  UpdateTitle,
+  UpdateDescription,
+  SaveTimeline,
+} from '../properties/helpers';
+import { NOTES_PANEL_WIDTH } from '../properties/notes_size';
+import { TIMELINE_TITLE, DESCRIPTION } from '../properties/translations';
+import { useCreateTimelineButton } from '../properties/use_create_timeline';
+import * as i18n from './translations';
+
+interface SaveTimelineButtonProps {
+  timelineId: string;
+  showOverlay: boolean;
+  toolTip?: string;
+  toggleSaveTimeline: () => void;
+  onSaveTimeline: SaveTimeline;
+  updateTitle: UpdateTitle;
+  updateDescription: UpdateDescription;
+}
+
+interface TimelineTitleAndDescriptionProps {
+  timelineId: string;
+  toggleSaveTimeline: () => void;
+  onSaveTimeline: SaveTimeline;
+  updateTitle: UpdateTitle;
+  updateDescription: UpdateDescription;
+}
+
+const Wrapper = styled(EuiModalBody)`
+  .euiFormRow {
+    max-width: none;
+  }
+
+  .euiFormControlLayout {
+    max-width: none;
+  }
+
+  .euiFieldText {
+    max-width: none;
+  }
+`;
+
+Wrapper.displayName = 'Wrapper';
+
+export const TimelineTitleAndDescription = React.memo<TimelineTitleAndDescriptionProps>(
+  ({ timelineId, toggleSaveTimeline, onSaveTimeline, updateTitle, updateDescription }) => {
+    const timeline = useShallowEqualSelector((state) =>
+      timelineSelectors.selectTimeline(state, timelineId)
+    );
+
+    const { description, isSaving, savedObjectId, title, timelineType } = timeline;
+
+    const handleClick = useCallback(() => {
+      onSaveTimeline({
+        ...timeline,
+        id: timelineId,
+      });
+    }, [onSaveTimeline, timeline, timelineId]);
```

Comment:
Consider closing the modal when the `Save` button is clicked:

![close-when-saved](https://user-images.githubusercontent.com/4459398/97201367-fd1cc400-1777-11eb-90b1-c9cb75a19e29.gif)